src/azure-cli-core/azure/cli/core/auth/msal_credentials.py

```diff
@@ -147,9 +162,13 @@ def __init__(self, client_id=None, resource_id=None, object_id=None):
             managed_identity = SystemAssignedManagedIdentity()
         self._msal_client = ManagedIdentityClient(managed_identity, http_client=requests.Session())
 
-    def acquire_token(self, scopes, **kwargs):
+    def acquire_token(self, scopes, data=None, **kwargs):
         logger.debug("ManagedIdentityCredential.acquire_token: scopes=%r, kwargs=%r", scopes, kwargs)
 
+        if data is not None:
+            from azure.cli.core.azclierror import AuthenticationError
+            raise AuthenticationError("VM SSH currently doesn't support managed identity.")
```

Comment:
This detection was in `adal_authentication.MSIAuthenticationWrapper`:

https://github.com/Azure/azure-cli/blob/7811532ba08c4a3e1598ad5a692c57ee551e7a6e/src/azure-cli-core/azure/cli/core/auth/adal_authentication.py#L38-L39

It got dropped by https://github.com/Azure/azure-cli/pull/31577 without being migrated to `msal_credentials.ManagedIdentityCredential`.